### ISSUE TITLE
App registration fails (suspected push notification registration fail)

### ISSUE THREAD

**Issue by nodomain** [WhisperSystems/Signal-iOS#979]
Hi,
I am unable to register with Signal since I can not receive push notifications. Signal is also missing from the app list for push notifications in the iOS settings app. Reinstall did not help. Error was already present on my iPhone 5S. Yesterday I switched to an iPhone 6S and the error persists. 
I am using public beta 9.2. 

Thanks for advice.

**Comment by FredericJacobs** [comment]
What do you mean by "since I can not receive push notifications"?

**Comment by nodomain** [comment]
I am referring to #761 and I just saw that I posted there about a month ago. Sorry, it was late last night ;-)
But basically the problem still persists and I can not register for Signal because the app does not receive the initial push notification during code verification.

**Comment by FredericJacobs** [comment]
What initial push notification? You mean the code? It's supposed to be an SMS.

**Comment by nodomain** [comment]
I receive that. But after entering the code nothing happens which leads me to the conclusion that something is wrong with push notifications. Signal is not in the list of apps for push notifications, neither am I asked for permission after reinstallation.

**Comment by FredericJacobs** [comment]
This is really strange. Would you mind helping us debug this by installing a beta on your device? It should only take a few taps.

**Comment by nodomain** [comment]
Of course, please :)

**Comment by FredericJacobs** [comment]
Can you email me your AppleID to fred@whispersystems.org?

**Comment by nodomain** [comment]
Done

**Comment by chronoslynx** [comment]
I'm seeing this on iOS 9.1 with a fresh install off of master (and the app store) on an iPhone 6. Between Installation and submitting my verification code I see the following debug output:

```
2015-12-03 08:49:38.174 Signal[7733:3043761] iRate verbose logging enabled.
2015-12-03 08:49:39:330 Signal[7733:3043761] No previous version found. Possibly first launch since install.
2015-12-03 08:49:39.559 Signal[7733:3043761] Presenting view controllers on detached view controllers is discouraged <SignalsViewController: 0x144583680>.
2015-12-03 08:49:39.628 Signal[7733:3043761] Failed to set (keyPath) user defined inspected property on (UITextField): [<UITextField 0x144715aa0> setValue:forUndefinedKey:]: this class is not key value coding-compliant for the key keyPath.
2015-12-03 08:49:39.655 Signal[7733:3043761] iRate did not prompt for rating because the app was first used less than 15 days ago
2015-12-03 08:49:39.655 Signal[7733:3043833] iRate is checking http://itunes.apple.com/US/lookup?id=874139669 to retrieve the App Store details...
2015-12-03 08:49:39.856 Signal[7733:3043833] iRate found that the application bundle ID (<MY BUNDLE>) does not match the bundle ID of the app found on iTunes (org.whispersystems.signal) with the specified App Store ID (874139669)
2015-12-03 08:49:40.569 Signal[7733:3043761] Can't find keyplane that supports type 5 for keyboard iPhone-PortraitChoco-PhonePad; using 3959308031_PortraitChoco_iPhone-Complex-Pad_Default
2015-12-03 08:49:40.630 Signal[7733:3043761] Unbalanced calls to begin/end appearance transitions for <SignalsNavigationController: 0x144856a00>.
2015-12-03 08:49:58.709 Signal[7733:3043761] Can't find keyplane that supports type 4 for keyboard iPhone-PortraitChoco-NumberPad; using 1336863583_PortraitChoco_iPhone-Simple-Pad_Default
```

**Comment by riyapenn** [comment]
Have you enabled notifications and background app refresh for Signal under your iPhone's settings?  Also support page here: http://support.whispersystems.org/hc/en-us/articles/213132457-Are-you-having-trouble-verifying-on-Signal-

**Comment by nodomain** [comment]
After activating background app refresh I was asked for notifications and enabled those as well. Verification worked.
Perhaps it is a good idea to display an info box that the user needs to enable background app refresh. My personal policy for my iOS devices is to enable background app refresh only for some apps and not by default.

**Comment by Liebach** [comment]
Freshly installed Signal from the App Store on iOS 9.2/iPhone 5s I'm seeing the exact same issue originally described by nodomain. Under Settings > Signal everything is enabled, under Settings > Notifications there's no entry for Signal to be found.

Rebooting the phone, reinstalling Signal, uninstalling Signal, reboot phone, install Signal again — nothing helps, no Signal under Settings > Notifications.

**Comment by FredericJacobs** [comment]
A fix has been implemented and is being tested. Email me your AppleID if you want to beta-test the fix: fred@whispersystems.org

**Comment by kunickiaj** [comment]
+1

Interested in doing some development, but can't actually verify on iOS 9.2 + iPhone 6.

**Comment by Liebach** [comment]
I got it to work.

I think that the deciding factor was that I uninstalled Signal, shut down all other apps and restarted the phone. Then, before doing anything else, I opened the app store, installed signal, and went through registration, and it just worked. Maybe I rebooted the phone after installing Signal, before registering, can't remember, sorry.

But it works now, and it seems a bit strange that the steps above was what made it work. Alternatively it's a heisenbug and it was just a matter of trying enough times for it to work.
